lib/api/apiUtils/bucket/bucketCreation.js

```diff
@@ -165,7 +165,9 @@ function createBucket(authInfo, bucketName, headers,
     const ownerDisplayName =
         authInfo.getAccountDisplayName();
     const creationDate = new Date().toJSON();
-    const objectLockEnabled = headers['x-amz-bucket-object-lock-enabled'];
+    const headerObjectLock = headers['x-amz-bucket-object-lock-enabled'];
+    const objectLockEnabled = headerObjectLock === 'true'
+        || headerObjectLock === true;
```

Comment:
I don't think this can be a boolean. All header values are strings.

Reply:
You're right, just verified with AWS cli and sdk as well; unit test failures tricked me into it but I just edited those tests so should be good now. :)